File: src/main/java/com/alealogic/singleproxy/repository/DesktopClientRepository.java
package com.alealogic.singleproxy.repository;

import com.alealogic.singleproxy.entity.DesktopClient;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface DesktopClientRepository extends JpaRepository<DesktopClient, Long> {

    Optional<DesktopClient> findByKey(String key);
}

File: src/main/java/com/alealogic/singleproxy/repository/DesktopClientRepository.kt
package com.alealogic.singleproxy.repository

import com.alealogic.singleproxy.entity.DesktopClient
import org.springframework.data.jpa.repository.JpaRepository

interface DesktopClientRepository : JpaRepository<DesktopClient?, Long?> {
    fun findByKey(key: String?): DesktopClient?

    fun findByUpdateInitiatorDesktopClientId(id: Long?): DesktopClient?

    fun findAllByActiveTrue(): MutableSet<DesktopClient>

    fun findByIpId(ipId: String): DesktopClient?

    fun findByCustomerId(customerId: Long): DesktopClient?
}
